.github/workflows/test.yml

```diff
@@ -0,0 +1,82 @@
+name: SmartDeviceLink Tests
+
+# This workflow is triggered on a push or pull request.
+on:
+    # Runs the workflow when someone pushes to any repository branch.
+    push:
```

Comment:
I can't comment on your comment for some reason so I'll just leave a comment here...

I can't figure out the formatting that allows you to remove the colon after the keyword is you have another keyword that does need the colon, so I am just going to leave it as is. I added a comment to the "pull_request" keyword explaining how it works.